gossip/ethapi_backend.go

```diff
@@ -379,3 +385,145 @@ func (b *EthAPIBackend) ServiceFilter(ctx context.Context, session *bloombits.Ma
 		}
 	*/
 }
+
+// CurrentEpoch returns current epoch number.
+func (b *EthAPIBackend) CurrentEpoch(ctx context.Context) idx.Epoch {
+	return b.svc.engine.GetEpoch()
+}
+
+// GetEpochStats returns epoch statistics.
+func (b *EthAPIBackend) GetEpochStats(ctx context.Context, requestedEpoch rpc.BlockNumber) (*sfctype.EpochStats, idx.Epoch, error) {
+	var epoch idx.Epoch
+	if requestedEpoch == rpc.PendingBlockNumber {
+		epoch = pendingEpoch
+	} else if requestedEpoch == rpc.LatestBlockNumber {
+		epoch = b.CurrentEpoch(ctx) - 1
+	} else {
+		epoch = idx.Epoch(requestedEpoch)
+	}
+	if epoch == b.CurrentEpoch(ctx) {
+		return nil, 0, errors.New("current epoch isn't sealed yet, request pending epoch")
+	}
+
+	return b.svc.store.GetEpochStats(epoch), epoch, nil
+}
+
+// GetValidationScore returns staker's ValidationScore.
+func (b *EthAPIBackend) GetValidationScore(ctx context.Context, stakerID idx.StakerID) (*big.Int, error) {
+	if !b.svc.store.HasSfcStaker(stakerID) {
+		return nil, nil
+	}
+	return new(big.Int).SetUint64(b.svc.store.GetActiveValidationScore(stakerID)), nil
+}
+
+// GetOriginationScore returns staker's OriginationScore.
+func (b *EthAPIBackend) GetOriginationScore(ctx context.Context, stakerID idx.StakerID) (*big.Int, error) {
+	if !b.svc.store.HasSfcStaker(stakerID) {
+		return nil, nil
+	}
+	return new(big.Int).SetUint64(b.svc.store.GetActiveOriginationScore(stakerID)), nil
+}
+
+// GetStakerPoI returns staker's PoI.
+func (b *EthAPIBackend) GetStakerPoI(ctx context.Context, stakerID idx.StakerID) (*big.Int, error) {
+	if !b.svc.store.HasSfcStaker(stakerID) {
+		return nil, nil
+	}
+	return new(big.Int).SetUint64(b.svc.store.GetStakerPOI(stakerID)), nil
+}
+
+// GetValidatingPower returns staker's ValidatingPower.
+func (b *EthAPIBackend) GetValidatingPower(ctx context.Context, stakerID idx.StakerID) (*big.Int, error) {
+	if !b.svc.store.HasSfcStaker(stakerID) {
+		return nil, nil
```

Comment:
same: nill error

Reply:
double check all the methods, many of which just return nil error.

Reply:
Don't worry, it's for forward compatibility

Reply:
If you wander why there's no error if value not found - then we explicitly return nil in this case, like other web3 methods (unless missing value is an error)

Reply:
common use:

if err := ...; err != nil {
  ...
}

add comment to the methods as the common wisdom may not hold. Minor anyway.